src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaParameterModel.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaParameterModel : IJavaResolvable
+	{
+		public string Name { get; }
+		public string Type { get; }
+		public string JniType { get; }
+		public bool IsNotNull { get; }
+		public string GenericType { get; }
+
+		public JavaMethodModel ParentMethod { get; }
+		public JavaTypeReference? TypeModel { get; private set; }
+		public bool IsParameterArray { get; private set; }
+		public string? InstantiatedGenericArgumentName { get; internal set; }
+
+		public JavaParameterModel (JavaMethodModel parent, string javaName, string javaType, string jniType, bool isNotNull)
+		{
+			ParentMethod = parent;
+			Name = javaName;
+			Type = javaType;
+			JniType = jniType;
+			IsNotNull = isNotNull;
+			GenericType = javaType;
+
+			if (Type.Contains ('<'))
```

Comment:
Not quite related, but found this horror while looking for something related:

```xml
<parameter name="sharedElements" type="android.util.Pair&lt;android.view.View, java.lang.String&gt;..." jni-type="[Landroid/util/Pair&lt;Landroid/view/View;Ljava/lang/String;&gt;;" />
```

The `//parameter/@jni-type` value is *wrong*, and contains generic info when it shouldn't… :frowning_face:

Reply:
What I was *trying* to determine is this: what about nested generic types?  How is that represented, vs. how *should* it be represented?

So I look:

```bash
% grep '<parameter.*type=".*&lt;.*&gt;\.' bin/BuildRelease/api/api-30.xml.in
```

and find the oddest result:

```xml
<parameter name="entries" type="java.util.Map.Entry&lt;? extends K, ? extends V&gt;..." jni-type="[Ljava/util/Map$Entry&lt;+TK;+TV;&gt;;" not-null="true" />
```

We have [`java.util.Map.Entry`](https://developer.android.com/reference/java/util/Map.Entry).  `Map<K,V>` is a generic type, and `Map.Entry` is a nested type.

…and now I see that I've been doing C# too long (?): I had *thought* that in Java, I would say `Map<String, String>.Entry`.  However, that's not valid for a method parameter.  Instead, it's as shown in the above XML!

```java
public static void m(Map.Entry<String, String> e) {}
```

which I did not expect.

Consequently, grabbing everything before the `<` appears to be fine!  (Entirely unexpected, but fine!)

TIL!